packages/sdk/src/util/logger.ts

```diff
@@ -1,2 +1,22 @@
-export type Logger = (message: string) => void;
+export type SimpleLogger = (message: string) => void;
 export const noopLogger = () => {};
+
+export interface IFullLogger {
+  error: SimpleLogger;
+  warn: SimpleLogger;
+  report: SimpleLogger;
+}
+
+export type Logger = SimpleLogger | IFullLogger;
+
+export function toFullLogger(l: Logger): IFullLogger {
+  if ('warn' in l) {
```

Comment:
This might need to be a bit more strict than duck-typing since you've got 3 methods all of which are actually used in services.

I'm thinking to the effect of `return typeof l === 'function' ? { report: l, error: m, ...  } : logger `

Reply:
Yeah that's much better!